ledger/sandbox-classic/src/main/scala/platform/sandbox/SandboxServer.scala

```diff
@@ -63,14 +63,19 @@ object SandboxServer {
 
   private val logger = ContextualizedLogger.get(this.getClass)
 
-  // FIXME: https://github.com/digital-asset/daml/issues/5164
-  // This should be made configurable
-  @silent("Sandbox_Classic in object EngineConfig is deprecated")
-  private[sandbox] val engineConfig: EngineConfig = EngineConfig.Sandbox_Classic
-
   // We memoize the engine between resets so we avoid the expensive
   // repeated validation of the sames packages after each reset
-  private val engine = new Engine(engineConfig)
+  private[this] var engine: Option[Engine] = None
+
+  private def getEngine(config: EngineConfig): Engine = synchronized {
+    engine match {
+      case Some(eng) if eng.config == config => eng
+      case _ =>
+        val eng = new Engine(config)
+        engine = Some(eng)
+        eng
+    }
+  }
```

Comment:
@SamirTalwar-DA what do you think about that ?

Reply:
Looks good to me. 😀